gmso/utils/misc.py

```diff
@@ -22,3 +23,21 @@ def _unyt_to_hashable_single(val):
         return tuple(val.in_base().value)
     else:
         return None
+
+
+def ensure_valid_dimensions(quantity_1: u.unyt_quantity,
+                            quantity_2: u.unyt_quantity) -> None:
+    """Ensure that the quantities provided as arguments have valid dimensions
```

Comment:
We need a better description for this method. Something like this suggested by @justinGilmer .

```suggestion
    This utility will verify that the two unyt quantities provided simplify to the same dimensionality. This ensures that each quantity can be converted to the desired unit system.
```